pkg/cluster/provider.go

```diff
@@ -0,0 +1,59 @@
+package cluster
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/watch"
+	"sigs.k8s.io/logical-cluster"
+)
+
+// Provider defines methods to retrieve, list, and watch fleet of clusters.
+// The provider is responsible for discovering and managing the lifecycle of each
+// cluster.
+//
+// Example: A Cluster API provider would be responsible for discovering and managing
+// clusters that are backed by Cluster API resources, which can live
+// in multiple namespaces in a single management cluster.
+type Provider interface {
+	Get(ctx context.Context, name logical.Name, opts ...Option) (Cluster, error)
+
+	// List returns a list of logical clusters.
+	// This method is used to discover the initial set of logical clusters
+	// and to refresh the list of logical clusters periodically.
+	List() ([]logical.Name, error)
```

Comment:
nit: I wonder if a context also makes sense here, and maybe in watch as well

Reply:
This is most likely backed by an informer, but doesn't have to. It could reach out to a remote system.

Added context.